### PR TITLE
(EAI-174) [Core] Support arbitrary MongoDB queries in PageStore

### PR DESCRIPTION
**Pull request by nlarew**
Jira: (EAI-174) [Core] Support arbitrary MongoDB queries in PageStore

### CODE REVIEW

File: packages/mongodb-rag-core/src/Page.ts [hunk 1]
```diff
@@ -54,26 +56,35 @@ export type PersistedPage = Page & {
   action: PageAction;
 };
 
+export type LoadPagesQuery = {
```
Comment (mongodben) [inline packages/mongodb-rag-core/src/Page.ts:59]: why support both this and LoadPageArgs?

isn't LoadPagesArgs a subset of LoadPagesQuery?

Comment (mongodben) [reply packages/mongodb-rag-core/src/Page.ts:59]: also maybe we could abstract this to a `Record<string, unknown>` here, and then in the makeMongoDbPageStore implementation use `Filter<PersistedPage>`. this way the Page type abstraction isn't tightly coupled to MongoDB.

File: packages/mongodb-rag-core/src/MongoDbPageStore.ts [hunk 2]
```diff
@@ -20,16 +20,13 @@ export function makeMongoDbPageStore({
   });
   const pagesCollection = db.collection<PersistedPage>("pages");
   return {
+    queryType: "mongodb",
```
Comment (cbush) [inline packages/mongodb-rag-core/src/MongoDbPageStore.ts:23]: If it's possible, I would export a special type:

```ts
export type MongoDBPageStore = PageStore & DatabaseConnection & {
  queryType: "mongodb";
  loadPages(args?: LoadPagesArgs<Filter<PersistedPage>>): Promise<PersistedPage[]>;
};
```
Then users can do
```ts
if (store.queryType === "mongodb") {
  (store as MongoDbPageStore).loadPages({ ... }); // VS Code helps with 'filter' type
}
```
Return that type from makeMongoDbPageStore.

Note that I imply LoadPagesArgs could be `LoadPagesArgs<T = unknown>` where T is the type of `query`. SHOULD work? (Might not)

**Review by cbush — APPROVED** [review]

File: packages/mongodb-rag-core/src/MongoDbPageStore.ts (hunk 2 above)
Comment (nlarew) [reply packages/mongodb-rag-core/src/MongoDbPageStore.ts:23]: Does work!

**Review by cbush — APPROVED** [review]

**Pull request merged**